File: cpp/Find Pivot Index.cpp
#include "common.h"

class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        vector<int> leftSum = {0};
        vector<int> rightSum = {0};

        for (int i=1; i< nums.size(); i++) {
            leftSum.push_back(nums[i-1]+ leftSum.back());
        }

        for (int i=nums.size()-2; i >= 0; i--) {
            rightSum.push_back(nums[i+1] + rightSum.back());
        }
        for (int i=0; i < nums.size(); i++) {
            if (leftSum[i] == rightSum[nums.size()-i-1]) {
                return i;
            }
        }

        return -1;
    }
};

File: java/Find Pivot Index.java

import java.util.Set;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;


class Solution {
    public int pivotIndex(int[] nums) {
        int right = IntStream.of(nums).sum();
        int left = 0;


        for (int i=0; i<nums.length; i++) {
            right -= nums[i];
            if (left == right) {
                return i;
            }
            left += nums[i];
        }

        return -1;

    }
}